cgroups/fs/apply_raw.go

```diff
@@ -118,21 +137,25 @@ func GetStats(systemPaths map[string]string) (*cgroups.Stats, error) {
 
 // Freeze toggles the container's freezer cgroup depending on the state
 // provided
-func Freeze(c *cgroups.Cgroup, state cgroups.FreezerState) error {
-	d, err := getCgroupData(c, 0)
+func (m *Manager) Freeze(state cgroups.FreezerState) error {
+	d, err := getCgroupData(m.Cgroups, 0)
 	if err != nil {
 		return err
 	}
 
-	c.Freezer = state
-
 	freezer := subsystems["freezer"]
```

Comment:
subsystems always has the "freezer" key

```golang
        subsystems = map[string]subsystem{
                "devices":    &DevicesGroup{},
                "memory":     &MemoryGroup{},
                "cpu":        &CpuGroup{},
                "cpuset":     &CpusetGroup{},
                "cpuacct":    &CpuacctGroup{},
                "blkio":      &BlkioGroup{},
                "perf_event": &PerfEventGroup{},
                "freezer":    &FreezerGroup{},
        }
```

I think data.path() does what you suggest.
https://github.com/docker/libcontainer/blob/master/cgroups/fs/apply_raw.go#L181